velox/functions/prestosql/tests/ArrayNormalizeTest.cpp

```diff
@@ -0,0 +1,153 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayNormalizeTest : public FunctionBaseTest {
+ protected:
+  void testExpr(
+      const VectorPtr& expected,
+      const std::vector<VectorPtr>& input) {
+    auto result =
+        evaluate<ArrayVector>("array_normalize(C0, C1)", makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  // Helper function to calculate p-norm based on its definition.
+  template <typename T>
+  T pNorm(const std::vector<T> values, T p) {
+    T sum = 0;
+    for (const auto& item : values) {
+      sum += pow(abs(item), p);
+    }
+    return pow(sum, 1.0 / p);
+  }
+};
+
+TEST_F(ArrayNormalizeTest, array_with_elements_zero) {
+  auto input = makeArrayVector<double>({{0.0, -0.0, 0.0}});
+  auto p = makeConstant(2.0, input->size());
+  auto expected = makeArrayVector<double>({{0.0, -0.0, 0.0}});
+  testExpr(expected, {input, p});
+}
+
+TEST_F(ArrayNormalizeTest, p_equals_0) {
+  auto input = makeArrayVector<double>({{1.0, -2.0, 3.0}});
+  auto p = makeConstant(0.0, input->size());
+  auto expected = makeArrayVector<double>({{1.0, -2.0, 3.0}});
+  testExpr(expected, {input, p});
+}
+
+TEST_F(ArrayNormalizeTest, p_less_than_1) {
+  auto input = makeArrayVector<double>({{1.0, -4.0, 9.0}});
+  auto p = makeConstant(0.5, input->size());
+  auto expected = makeArrayVector<double>({{1.0 / 36, -4.0 / 36, 9.0 / 36}});
+  testExpr(expected, {input, p});
+}
+
+TEST_F(ArrayNormalizeTest, p_equals_1) {
+  auto input = makeArrayVector<double>({{1, -2, 3}});
+  auto p = makeConstant(1.0, input->size());
+  auto expected = makeArrayVector<double>({{1.0 / 6, -2.0 / 6, 3.0 / 6}});
+  testExpr(expected, {input, p});
+}
+
+TEST_F(ArrayNormalizeTest, limits) {
+  auto input = makeArrayVector<double>({
+      {},
+      {0},
+      {1},
+      {std::numeric_limits<double>::infinity()},
+      {std::numeric_limits<double>::quiet_NaN()},
+      {std::numeric_limits<double>::lowest(), -1.0},
+      {std::numeric_limits<double>::max(), 1.0},
+  });
+  auto search = makeConstant(2.0, input->size());
+  auto expected = makeArrayVector<double>(
+      {{},
+       {0},
+       {1},
+       {std::numeric_limits<double>::quiet_NaN()},
+       {std::numeric_limits<double>::quiet_NaN()},
+       {-0.0, -0.0},
```

Comment:
I added a comment, this is basically due to pNorm becoming infinity so any values end up being 0. Minus sign is mostly academic because the input values are negative.